### DISCUSSION TITLE
sysctl অপটিমাইজেশন

### DISCUSSION THREAD

**Discussion by giscus** [ahmubashshir/ahmubashshir.github.io#47] [Comments]
# sysctl অপটিমাইজেশন

আমার তেমন লেখালেখির অভ্যাস নাই…. তো কাজের কথা শুরু করি। এই পোস্টের বিষয় হল sysctl টুল দিয়ে কিভাবে কার্নেলের মেমরি/আই.ও. রিলেটেড প্যারামিটার অপ্টিমাইজ করা যায় তা দেখানো।
vm.swappiness এই অপশন এর ভ্যালু কমায় দিলে কার্নেল যখন তখন সোয়াপ করবে না1। এইটার ডিফল্ট ভ্যালু 60 থাকে… তো যেইটা করতে হবে:
# /etc/sysctl.d/10-vm-swap.conf vm.swappiness = 10 `vm.swappiness` 0 করলে কার্নেল একবারে বিপদে না পড়লে সোয়াপ করবে না। vm.vfs_cache_pressure এই অপশন এর ভ্যালু ১০০ এর বেশি সেট করলে কার্নেল যত দ্রুত সম্ভব আইনোড ক্যাশ ক্লিয়ার করবে2। তবে এইটা বেশি দিলে সফটওয়্যার স্টার্ট নিতে সময় বেশি লাগতে পারে। ১০০০ এর উপরে সেট না করাই ভালো।

https://ahmubashshir.github.io/post/2020/11/07/sysctl-optimization/

**Comment by SharafatKarim** [comment]
/etc/sysctl.d/ directory আমার empty, কি কত্তাম?
